File: sumarhastan.py

import psycopg2
def  post(x,y):
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "jblue5queso",
            dbname = "tarea1"
            )
    except:
        print("Sin Conexion Exitosa\n")
            
    cursor = conexion.cursor()
    cursor.execute("INSERT INTO sumarhastan(n, suma) VALUES (%s, %s);", (x, y))
    conexion.commit()
    cursor.close()
    conexion.close()


print("Ingrese el valor final de la sumatoria: ")
a = int(input())
b=0
for i in range(1, a+1):
    b=b+i
print("La suma es: ", b)
post(a, b)

File: sumarhastan.m
clc; 
clear all; 
close all;
num = input('Ingrese el numero final de la sumatoria: ');
sum=0;
for x=0:1:num
  sum=sum+x;
end

fprintf('El resultado de la sumatoria es: %d\n', sum)

%Conectar con PGADMIN
params= cell(1,2);
params{1,1}=num;
params{1,2}=sum;

conn = pq_connect(setdbopts('dbname','tarea1','host','localhost',
'port','5432','user','postgres','password','jblue5queso'));
N=pq_exec_params(conn, "insert into sumarhastan(n,suma) values($1,$2);",params); %insertar datos en la tabla

%Texto
save('sumarhastan.txt','num','sum','-ascii');
